File: uikit/src/main/java/io/agora/chat/uikit/chat/viewholder/EaseCustomViewHolder.java
package io.agora.chat.uikit.chat.viewholder;

import android.view.View;


import androidx.annotation.NonNull;

import io.agora.chat.ChatClient;
import io.agora.chat.ChatMessage;
import io.agora.chat.uikit.interfaces.MessageListItemClickListener;
import io.agora.chat.uikit.manager.EaseConfigsManager;
import io.agora.chat.uikit.manager.EaseDingMessageHelper;
import io.agora.exceptions.ChatException;

public class EaseCustomViewHolder extends EaseChatRowViewHolder {

    public EaseCustomViewHolder(@NonNull View itemView, MessageListItemClickListener itemClickListener) {
        super(itemView, itemClickListener);
    }

    @Override
    protected void handleReceiveMessage(ChatMessage message) {
        if(!EaseConfigsManager.enableSendChannelAck()) {
            //此处不再单独发送read_ack消息，改为进入聊天页面发送channel_ack
            //新消息在聊天页面的onReceiveMessage方法中，排除视频，语音和文件消息外，发送read_ack消息
            if (!message.isAcked() && message.getChatType() == ChatMessage.ChatType.Chat) {
                try {
                    ChatClient.getInstance().chatManager().ackMessageRead(message.getFrom(), message.getMsgId());
                } catch (ChatException e) {
                    e.printStackTrace();
                }
                return;
            }
        }

        // Send the group-ack cmd type msg if this msg is a ding-type msg.
        EaseDingMessageHelper.get().sendAckMessage(message);
    }
}

File: ease-im-kit/src/main/kotlin/io/agora/uikit/feature/chat/viewholders/EaseCustomViewHolder.kt
package io.agora.uikit.feature.chat.viewholders

import android.view.View
import io.agora.uikit.common.ChatMessage
import io.agora.uikit.common.helper.EaseDingMessageHelper

class EaseCustomViewHolder(itemView: View) : EaseChatRowViewHolder(itemView) {
    override fun handleReceiveMessage(message: ChatMessage?) {
        super.handleReceiveMessage(message)
        message?.let {
            // Send the group-ack cmd type msg if this msg is a ding-type msg.
            EaseDingMessageHelper.get().sendAckMessage(it)
        }
    }
}